### PR TITLE
Expose Prometheus metrics for Vector

### PR DESCRIPTION
**Pull request by 4n4nd**
Update vector to v0.14.x
Use internal route for kafka sync

Resolves https://github.com/operate-first/apps/issues/820

### CODE REVIEW

File: vector/overlays/moc/zero/opf-observatorium/service.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  selector:
+    app: vector
+  ports:
+    - name: metrics
+      protocol: TCP
+      port: 80
+      targetPort: metrics
+    - name: graphql-api
+      protocol: TCP
+      port: 8000
```
Comment (tumido) [inline vector/overlays/moc/zero/opf-observatorium/service.yaml:16]: Is there a reason to remap the port numbers? Why not keep it consistent with the actually exposed ports?

Comment (4n4nd) [reply vector/overlays/moc/zero/opf-observatorium/service.yaml]: updated :+1:

**Review by HumairAK — APPROVED** [review]
/lgtm
/approve

**Pull request merged**